### DISCUSSION TITLE
A little gap between this repo and the official result.

### DISCUSSION THREAD

**Discussion by TeaPoly** [jzi040941/PercepNet#40] [Q&A]
Thanks for this great repo.
I have dump audio from https://www.amazon.science/blog/how-amazon-chimes-challenge-winning-noise-cancellation-works.
`percepnet-noisy.wav` and `percepnet-clean.wav` is probe and target audio. `percepnet.oracle_bands.wav` is process only using band from Amazon. The result is very close to your repo `test-bands.pcm`. But the band gain and pitch strength result `percepnet-oracle-pitch.wav` from Amazon is different to the result `test-pitch.pcm` from your repo. It sounds more lower volume and has a little roughness.
Maybe we can find clues together.

Here is all audio files:
https://drive.google.com/file/d/1Kf6R0nOGI1LyLZ1wDQlz90LXSXyr92AN/view?usp=sharing

**Comment by jzi040941** [comment]
Hello, Thank you for reporting the difference between result from my repo and offical Amazon result.
I've test few things and finally get the most similar output at branch `filterstrength`
currently, it's on pull requests
please let me know your opinion for this version

**Reply by jzi040941** [reply]
https://drive.google.com/file/d/1VcqJDZt2XkAhbEdl348baXfI6GYzdjB4/view?usp=sharing

this is the output of branch `filterstrength` from the input files that you share.

**Reply by TeaPoly** [reply]
Great job. It sounds good.

**Comment by TeaPoly** [comment]
Have you test DNS 2020 synthetic test set (`no_reverb`/`with_reverb`)  PESQ (WB/NB) with your repo?

**Reply by TeaPoly** [reply]
Here is `no_reverb` synthetic  test set  PESQ1([ITU-T P.862.1](https://www.itu.int/rec/T-REC-P.862/en)), PESQ2([ITU-T P.862.2](https://www.itu.int/rec/T-REC-P.862/en)) result from this repo frontend with an oracle (ideal ratio model) result. 

Baseline result from [Real-time Denoising and Dereverberation with Tiny Recurrent U-Net](https://arxiv.org/abs/2102.03207).

<html>
<body>
<!DOCTYPE html><figure class="md-table-fig" cid="n2" mdtype="table" style="box-sizing: border-box; margin: 1.2em 0px; overflow-x: auto; max-width: calc(100% + 16px); padding: 0px; cursor: default; color: rgb(51, 51, 51); font-family: &quot;Avenir Next LT Pro&quot;, Lato, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-style: initial; text-decoration-color: initial;">

  |   |   | no_reverb |  
-- | -- | -- | -- | --
  | Size(M/MB) | RT | PESQ1 | PESQ2
Noisy | - | Yes | 2.45 | 1.58
NSnet | 1.2/4.84 | Yes | 2.68 | 1.81
ConvTasNet | 5.08/19.38 | No | - | 2.73
PoCoNet1 | 50/190.73 | No | - | 2.71
PoCoNet2 | 50/190.73 | No | - | 2.75
DCCRN-E | 3.7/14.11 | Yes | 3.27 | -
DCCRN-CL | 3.7/14.11 | No | 3.26 | -
TRU-NET(FP32) | 0.38/1.45 | Yes | 3.36 | 2.86
TRU-NET(INT8) | 0.38/0.36 | Yes | 3.35 | 2.84
Percepnet-Oracle | - | - | 3.37 | 2.60

</figure><br class="Apple-interchange-newline">
</body>
</html>

**Reply by TeaPoly** [reply]
I think post-filtering maybe degrade PESQ2 result. I will try more experiment.

**Reply by TeaPoly** [reply]
Add Percepnet-Oracle without post-filtering result. 
Post-filtering will get degraded PESQ result which mentioned in original [paper](https://arxiv.org/pdf/2008.04259.pdf).

  |   |   | no_reverb |  
-- | -- | -- | -- | --
  | Size(M/MB) | RT | PESQ1 | PESQ2
Noisy | - | Yes | 2.45 | 1.58
NSnet | 1.2/4.84 | Yes | 2.68 | 1.81
ConvTasNet | 5.08/19.38 | No | - | 2.73
PoCoNet1 | 50/190.73 | No | - | 2.71
PoCoNet2 | 50/190.73 | No | - | 2.75
DCCRN-E | 3.7/14.11 | Yes | 3.27 | -
DCCRN-CL | 3.7/14.11 | No | 3.26 | -
TRU-NET(FP32) | 0.38/1.45 | Yes | 3.36 | 2.86
TRU-NET(INT8) | 0.38/0.36 | Yes | 3.35 | 2.84
Percepnet-Oracle | - | - | 3.37 | 2.60
Percepnet-Oracle(no-pf) | - | - | 3.58 | 3.03
